samples/net/sockets/echo_client/boards/frdm_mcxw71.conf

```diff
@@ -0,0 +1,42 @@
+CONFIG_REQUIRES_FULL_LIBC=y
+
+CONFIG_COUNTER=y
+
+# Disable TCP and IPv4 (TCP disabled to avoid heavy traffic)
+CONFIG_NET_TCP=n
```

Comment:
Board overlays shouldn't copy generic configuration, just provide board-specific tweaks. IMHO the build should still rely on `overlay-ot.conf` in general, just provide config adjustments here.

Reply:
If I compile with `-DCONF_FILE="prj.conf overlay-ot.conf"` it doesn't take `boards/frdm_mcxw71.conf` automatically anymore.

Reply:
I believe I have to pass the `boards/frdm_mcxw71.conf` too in `DCONF_FILE`
https://docs.zephyrproject.org/latest/build/kconfig/setting.html#the-initial-configuration

Reply:
I updated the conf file

Reply:
> If I compile with -DCONF_FILE="prj.conf overlay-ot.conf" it doesn't take boards/frdm_mcxw71.conf automatically anymore.

That's because `DCONF_FILE` overrides the entire configuration, to apply overlays you should use `EXTRA_CONF_FILE` - that way Zephyr's build system applies configurations from all sources.
So building like `west build -b frdm_mcxw71 -- -DEXTRA_CONF_FILE=overlay-ot.conf` will merge all needed configs (prj.conf, overlay and board-specific overlay):
![image](https://github.com/user-attachments/assets/7778d2f9-2336-4137-9a3a-1fc5f23ecd56)